lib/pengine/complex.c

```diff
@@ -559,6 +559,12 @@ warn_about_deprecated_classes(pe_resource_t *rsc)
                      "Support for Upstart resources (such as %s) is deprecated "
                      "and will be removed in a future release of Pacemaker",
                      rsc->id);
+
+    } else if (pcmk__str_eq(std, PCMK_RESOURCE_CLASS_NAGIOS, pcmk__str_none)) {
+        pe_warn_once(pe_wo_nagios,
+                     "Support for Nagios resources (such as %s) is deprecated "
+                     "and will be removed in a future release of Pacemaker",
+                     rsc->id);
```

Comment:
I don't think a release number would help (it would be 3.0.0 in this case), since without searching further it still doesn't give them any idea of how much time they have. Also, I suspect leaving it vague will light a fire under someone more than giving them a deadline they can wait until the last minute to deal with.